emukit/model_wrappers/gpy_model_wrappers.py

```diff
@@ -209,3 +209,102 @@ def get_covariance_between_points(self, X1: np.ndarray, X2: np.ndarray) -> np.nd
         :return: An array of shape n_points x 1 of posterior covariances between X1 and X2
         """
         return self.gpy_model.posterior_covariance_between_points(X1, X2)
+
+
+class GPyModelWrapperMCMC(IModel, IDifferentiable, IPriorHyperparameters):
```

Comment:
There is no need for a separate wrapper here, just add implementation of IPriorHyperparameters to the original GPy wrapper, and make `n_samples=None` by default

Reply:
just merged both.

Reply:
with the new structure, there is no need of having `n_samples` here, as this is now directly specified when the method is called.